mvc/src/main/java/com/interface21/webmvc/servlet/mvc/tobe/AnnotationHandlerMapping.java

```diff
@@ -14,16 +20,53 @@ public class AnnotationHandlerMapping {
     private final Object[] basePackage;
     private final Map<HandlerKey, HandlerExecution> handlerExecutions;
 
-    public AnnotationHandlerMapping(final Object... basePackage) {
+    public AnnotationHandlerMapping(Object... basePackage) {
         this.basePackage = basePackage;
         this.handlerExecutions = new HashMap<>();
     }
 
-    public void initialize() {
+    public void initialize() throws Exception {
         log.info("Initialized AnnotationHandlerMapping!");
+
+        for (Object basePackage : basePackage) {
+            Reflections reflections = new Reflections(basePackage.toString());
+            Set<Class<?>> controllerClasses = reflections.getTypesAnnotatedWith(Controller.class);
+            for (Class<?> controllerClass : controllerClasses) {
+                registerRequestMappings(controllerClass);
+            }
+        }
     }
 
-    public Object getHandler(final HttpServletRequest request) {
-        return null;
+    private void registerRequestMappings(Class<?> controllerClass) throws Exception {
+        Method[] methods = controllerClass.getDeclaredMethods();
+        for (Method method : methods) {
+            if (method.isAnnotationPresent(RequestMapping.class)) {
+                RequestMapping requestMapping = method.getAnnotation(RequestMapping.class);
+                String url = requestMapping.value();
+                RequestMethod[] requestMethods = requestMapping.method();
+                registerHandlerExecutions(controllerClass, method, url, requestMethods);
+            }
+        }
+    }
+
+    private void registerHandlerExecutions(Class<?> controllerClass, Method method, String url,
+                                           RequestMethod[] requestMethods) throws Exception {
+        for (RequestMethod requestMethod : requestMethods) {
+            HandlerKey handlerKey = new HandlerKey(url, requestMethod);
+            Constructor<?> constructor = controllerClass.getDeclaredConstructor();
+            constructor.setAccessible(true);
+            Object controllerInstance = constructor.newInstance();
+            handlerExecutions.put(handlerKey, new HandlerExecution(controllerInstance, method));
+        }
+    }
+
+    public Object getHandler(HttpServletRequest request) {
+        String requestURI = request.getRequestURI();
+        RequestMethod requestMethod = RequestMethod.valueOf(request.getMethod().toUpperCase());
```

Comment:
```suggestion
        RequestMethod requestMethod = RequestMethod.valueOf(request.getMethod());
```

객체지향 생활체조
4. 한 줄에 점을 하나만 찍는다.

그리고 매번 클라이언트가 대문자로 변환해서 넘겨주는 것보다, RequestMethod 내부에서 직접 처리해주는게 사용하는 입장에서 편하지 않을까요?

Reply:
`RequestMethod`를 사용하는 사용자가 더 편하게 사용할 수 있군요 😀 `RequestMethod `에 문자열로부터 RequestMethod를 반환하는 정적 메소드를 추가로 만들었습니다 !